src/app/shared/guards/pending-changes-guard.ts

```diff
@@ -11,6 +12,9 @@ export interface PendingChangesComponent {
   isDirty(): boolean,
 }
 
+const dialogResponse = new BehaviorSubject<'pending' | 'accepted' | 'declined'>('pending');
```

Comment:
That's a kind of global variable? 
There should not be global variable in the app, the shared resources are supposed to use the angular DI mechanis for a similar results, in cleaner and testable way.

Reply:
Must be service. It was done quickly more for you to look this "proof of concept" solution